datafusion/src/physical_plan/datetime_expressions.rs

```diff
@@ -434,6 +438,11 @@ mod tests {
                 "week",
                 "2019-12-30T00:00:00.000000Z",
             ),
+            (
+                "2020-01-01T13:42:29.190855Z",
```

Comment:
I recommend at least one other test that shows a quarter other than `Q1` -- for example 

```
 "2020-05-023T13:42:29.190855Z",
```

I would expect the result to be `2020-04-01`.

FYI you can test expected answers in `postgres` using `pswl` with queries such as

```sql
alamb=# select date_trunc('quarter', cast('2020-05-023T13:42:29.190855Z' as timestamp));
     date_trunc      
---------------------
 2020-04-01 00:00:00
(1 row)

alamb=# select date_trunc('quarter', cast('2020-10-023T13:42:29.190855Z' as timestamp));
     date_trunc      
---------------------
 2020-10-01 00:00:00
(1 row)
```